engineering/conventions-go.md

```diff
@@ -69,3 +69,9 @@ A blank line is usually added after the end of any control structure, too.
 ### Naming
 
 * Prefer full names (e.g. `Repository`, not `Repo`).
+
+## Configuration
+
+* Use environment variables for configuration.
+* You can use the [envconfig](https://github.com/kelseyhightower/envconfig) package to parse the environment into a configuration struct.
```

Comment:
I think an exception could be made for some user-facing CLI applications like `bblfshctl`.

Image a case when you have a local and a remote `bblfshd` and you have to set 3 environment variables for each call (or open two terminals) to switch servers. It's easier to provide a config to the CLI.

Note that my point is not related to `bblfshd` itself, since it's easier to configure dockerized applications via environment variables.

As a middle ground, there is a project called [viper](https://github.com/spf13/viper) that allows setting CLI flags from env/config interchangeably.

Reply:
Environment variables should be present in any case. This allows to set these either globally in your machine or in a given session, set them in a Kubernetes pod meant to be used interactively, etc.

We have discussed, and I think with no definite conclusion yet, about CLI flags being present in some cases _in addition to_ environment variables. This might be useful, specially if these are documented together in `--help` just as viper, go-flags and others do. In go-flags you can do it with the `env` struct tag.

Reply:
@dennwc  go-flags, supports envconfig too. Personally, I don't like at all viper or cobra.

@smola  Can you replace `You can use` with `You MUST use`?

Reply:
@mcuadros Any specific concerns about cobra except that it's easier to use structs with field tags?